x-pack/plugins/security_solution/cypress/e2e/dashboards/upgrade_risk_score.cy.ts

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { getNewRule } from '../../objects/rule';
+import {
+  getRiskScoreLatestTransformId,
+  getRiskScorePivotTransformId,
+  RiskScoreEntity,
+  RISK_SCORE_INSTALLATION_SUCCESS_TOAST,
+  UPGRADE_HOST_RISK_SCORE_BUTTON,
+  UPGRADE_USER_RISK_SCORE_BUTTON,
+  UPGRADE_CANCELLATION_BUTTON,
+  UPGRADE_CONFIRMARION_MODAL,
+  RISK_SCORE_DASHBOARDS_INSTALLATION_SUCCESS_TOAST,
+} from '../../screens/entity_analytics';
+import { createCustomRuleEnabled } from '../../tasks/api_calls/rules';
+import { cleanKibana } from '../../tasks/common';
+import { login, visit } from '../../tasks/login';
+import {
+  clickUpgradeRiskScore,
+  clickUpgradeRiskScoreCancel,
+  clickUpgradeRiskScoreConfirmed,
+  deleteRiskScore,
+  installLegacyRiskScoreModule,
+  interceptUpgradeRiskScoreModule,
+  waitForUpgradeRiskScoreModule,
+} from '../../tasks/risk_scores';
+import { findSavedObjects } from '../../tasks/risk_scores/saved_objects';
+import { getTransformState } from '../../tasks/risk_scores/transforms';
+import { ENTITY_ANALYTICS_URL } from '../../urls/navigation';
+
+const spaceId = 'default';
+
+describe('Upgrade risk scores', () => {
+  before(() => {
+    cleanKibana();
+    login();
+    createCustomRuleEnabled(getNewRule(), 'rule1');
+  });
+
+  beforeEach(() => {
+    deleteRiskScore({ riskScoreEntity: RiskScoreEntity.host, spaceId, deleteAll: true });
+    deleteRiskScore({ riskScoreEntity: RiskScoreEntity.user, spaceId, deleteAll: true });
+    installLegacyRiskScoreModule(RiskScoreEntity.host, spaceId);
+    installLegacyRiskScoreModule(RiskScoreEntity.user, spaceId);
+    visit(ENTITY_ANALYTICS_URL);
+  });
+
+  afterEach(() => {
+    deleteRiskScore({ riskScoreEntity: RiskScoreEntity.host, spaceId, deleteAll: true });
+    deleteRiskScore({ riskScoreEntity: RiskScoreEntity.user, spaceId, deleteAll: true });
+  });
+
+  it('shows upgrade host risk button', () => {
+    cy.get(UPGRADE_HOST_RISK_SCORE_BUTTON).should('be.visible');
+  });
+
+  it('should show a confirmation modal for upgrading host risk score', () => {
+    clickUpgradeRiskScore(RiskScoreEntity.host);
+    cy.get(UPGRADE_CONFIRMARION_MODAL(RiskScoreEntity.host)).should('exist');
+    clickUpgradeRiskScoreCancel();
+  });
+
+  it('display a link to host risk score Elastic doc', () => {
+    clickUpgradeRiskScore(RiskScoreEntity.host);
+
+    cy.get(UPGRADE_CANCELLATION_BUTTON)
+      .get(`${UPGRADE_CONFIRMARION_MODAL(RiskScoreEntity.host)} a`)
+      .then((link) => {
+        expect(link.prop('href')).to.eql(
+          `https://www.elastic.co/guide/en/security/current/${RiskScoreEntity.host}-risk-score.html`
+        );
+      });
+
+    clickUpgradeRiskScoreCancel();
```

Comment:
A test never should finish with an action. What is going to happen if the assertion fails and the `clickUpgradeRiskScoreCancel` is not executed?